FirebaseAuth/Sources/Swift/AuthProvider/OAuthProvider.swift

```diff
@@ -103,12 +103,28 @@ import Foundation
   /// - Parameter providerID: The provider ID associated with the Auth credential being created.
   /// - Parameter accessToken: The access token associated with the Auth credential be created
   /// - Returns: An AuthCredential.
+  @available(swift, introduced: 100.0)
   @objc(credentialWithProviderID:accessToken:)
+  public static func credential(withProviderID providerID: NSString,
+                                accessToken: NSString) -> OAuthCredential {
+    return OAuthCredential(withProviderID: providerID as String, accessToken: accessToken as String)
+  }
+
+  @available(
+    *,
+    deprecated,
+    message: "Use `credential(providerID: AuthProviderID, accessToken: String) -> OAuthCredential` instead."
+  )
   public static func credential(withProviderID providerID: String,
                                 accessToken: String) -> OAuthCredential {
     return OAuthCredential(withProviderID: providerID, accessToken: accessToken)
   }
 
+  public static func credential(providerID: AuthProviderID,
+                                accessToken: String) -> OAuthCredential {
```

Comment:
@paulb777, fyi, I'm opting for the new API to drop the "with" bit because I don't think it improves the clarity of the signature. I think it was a leftover from the ObjC translation. LMK if you have any concerns.

Reply:
sgtm